pkgs/os-specific/linux/wireguard/default.nix

```diff
@@ -7,29 +7,25 @@ assert stdenv.lib.versionOlder kernel.version "5.6";
 
 stdenv.mkDerivation rec {
   pname = "wireguard";
-  version = "1.0.20200401";
+  version = "1.0.20200413";
 
   src = fetchzip {
     url = "https://git.zx2c4.com/wireguard-linux-compat/snapshot/wireguard-linux-compat-${version}.tar.xz";
-    sha256 = "1q4gfpbvbyracnl219xqfz5yqfc08i6g41z6bn2skx5x8jbll3aq";
+    sha256 = "11dpw1inszbc3qjcfnap74kgjxkyyrx90vxv6wmsgkbp8lsl4p66";
   };
 
-  preConfigure = ''
-    cd src
-    sed -i '/depmod/,+1d' Makefile
-  '';
-
   hardeningDisable = [ "pic" ];
 
   KERNELDIR = "${kernel.dev}/lib/modules/${kernel.modDirVersion}/build";
-  INSTALL_MOD_PATH = "\${out}";
-
-  NIX_CFLAGS = ["-Wno-error=cpp"];
 
-  nativeBuildInputs = [ perl ] ++ kernel.moduleBuildDependencies;
+  nativeBuildInputs = [ perl bc ] ++ kernel.moduleBuildDependencies;
 
+  preBuild = "cd src";
   buildFlags = [ "module" ];
-  installTargets = [ "module-install" ];
+
+  installFlags = [ "-C" KERNELDIR "M=$(PWD)" "WIREGUARD_VERSION=${version}"];
+  installTargets = [ "modules_install" ];
+  INSTALL_MOD_PATH = placeholder "out";
```

Comment:
n00b question: what does `placeholder` do?

Reply:
Nix will replace it by the actual store path of the "out" output (in this case there is only one output anyway which is "out" by default, so it just means the store path of the package). Its somewhat documented in the [nix manual](https://nixos.org/nix/manual/) (search for placeholder there).

Reply:
thx! :sparkles: